core/vote/vote_manager.go

```diff
@@ -0,0 +1,217 @@
+package vote
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth/downloader"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type (
+	getHighestJustifiedHeaderFunc func(chain consensus.ChainHeaderReader, header *types.Header) *types.Header
+)
+
+// VoteManager will handle the vote produced by self.
+type VoteManager struct {
+	mux *event.TypeMux
```

Comment:
TypeMux is deprecated.

Reply:
Correct. But here the event needed is sent from mux of original downloader, and the original download used mux such that we're unable to use feed which is mismatch the original downloader event mux